frontend/src/utils/exportUtils.ts

```
@@ -0,0 +1,367 @@
+      case "csv":
+        exportToCSV(data, `${filename}.csv`, onProgress);
+        break;
+      case "json":
+        exportToJSON(data, `${filename}.json`, options, onProgress);
+        break;
+      case "xlsx":
+        exportToXLSX(data, `${filename}.xlsx`, "Dashboard Data", onProgress);
+        break;
+      case "pdf":
+        // For PDF, we expect the element ID to be in options.metadata.elementId
+        const elementId = options.metadata?.elementId || "dashboard-container";
```

Biome (2.5.0)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)